### PR TITLE
Offd num cols fix

### PR DESCRIPTION
**Pull request by victorapm**
This PR updates the `offd-num-cols-fix` branch wrt master. 

@pbauman can you take a look at this branch and confirm it's good to go?

PS: I initially set the base branch to `offd-num-cols-fix` here, but the number of changes to be reviewed was huge. So, I updated the base branch to `master`. After this PR is merged into master, we should close #726 as well

### CODE REVIEW

File: src/parcsr_mv/par_csr_matop_device.c [hunk 1]
```diff
@@ -1384,63 +1420,18 @@ hypre_ParCSRMatrixDropSmallEntriesDevice( hypre_ParCSRMatrix *A,
...
-                    HYPRE_MEMORY_HOST, HYPRE_MEMORY_DEVICE);
-   }
+   hypre_ParCSRMatrixCompressOffdMapDevice(A);
 
    if (type != 0)
    {
       hypre_TFree(elmt_tols_diag, HYPRE_MEMORY_DEVICE);
       hypre_TFree(elmt_tols_offd, HYPRE_MEMORY_DEVICE);
    }
+<<<<<<< HEAD
```
Comment (rfalgout) [inline src/parcsr_mv/par_csr_matop_device.c]: Looks like a merge conflict that needs to be resolved.

**Review by rfalgout — APPROVED** [review]
Looks good, though I did not dive into the details of the fix itself.  Thanks!

File: src/parcsr_mv/par_csr_matop_device.c (hunk 1 above)
Comment (victorapm) [reply src/parcsr_mv/par_csr_matop_device.c]: Ooops! Thanks for catching this

**Comment by liruipeng** [comment]
Thanks @victorapm . Are there any "real" changes on top of #726? If not, I guess I will just approve.

**Comment by victorapm** [comment]
Yes, Rui Peng! No "real" changes in this PR, I've just updated it with master so we can close this.

We are just missing @pbauman's or @noelchalmers' confirmation that this fixes the performance issues they saw with large node counts.

**Review by liruipeng — APPROVED** [review]

**Pull request merged**